src/config/config.rs

```diff
@@ -57,20 +57,37 @@ impl Config {
     types
   }
 
+  pub fn recurse_type<'a>(&'a self, type_of: &str, mut types: HashSet<&'a String>) -> HashSet<&'a String> {
```

Comment:
consider accepting `&mut HashSet<&'a String>` instead of cloning the hashmap since it's involves a lot of additional CPU cycles depending on size of the map

Reply:
Done